tests/integration/test_s3.py

```diff
@@ -85,6 +86,53 @@ def test_s3_put_object_notification():
     s3_client.delete_bucket(Bucket=TEST_BUCKET_WITH_NOTIFICATION)
 
 
+def generate_large_file(size):
+    # https://stackoverflow.com/questions/8816059/create-file-of-particular-size-in-python
+    filename = 'large_file_%s' % uuid.uuid4()
+    f = open(filename, 'wb')
+    f.seek(size - 1)
+    f.write(b'\0')
+    f.close()
+    return open(filename, 'r')
+
+
+def test_s3_upload_fileobj_with_large_file_notification():
+
+    s3_client = aws_stack.connect_to_service('s3')
+    sqs_client = aws_stack.connect_to_service('sqs')
+
+    # create test queue
+    queue_url = sqs_client.create_queue(QueueName=TEST_QUEUE_FOR_BUCKET_WITH_NOTIFICATION)['QueueUrl']
+    queue_attributes = sqs_client.get_queue_attributes(QueueUrl=queue_url, AttributeNames=['QueueArn'])
+
+    # create test bucket
+    s3_client.create_bucket(Bucket=TEST_BUCKET_WITH_NOTIFICATION)
+    s3_client.put_bucket_notification_configuration(Bucket=TEST_BUCKET_WITH_NOTIFICATION,
+                                                    NotificationConfiguration={'QueueConfigurations': [
+                                                        {'QueueArn': queue_attributes['Attributes']['QueueArn'],
+                                                         'Events': ['s3:ObjectCreated:*']}]})
+
+    large_file = generate_large_file(75000000)  # Has to be larger than 64MB to be broken up into a multipart upload
```

Comment:
To ensure the test does not leave any files behind (even in the case that an exception is thrown somewhere in the code below), can we please either:
1. wrap lines 116-126 in a `try`-`finally` block and perform the clean-up in `finally`,
or
2. instead of creating an actual file on disk, change `generate_large_file(..)` to return an `io.BytesIO` that creates a file object in memory

If the S3 SDK supports in-memory file objects, I'd actually prefer option (2), to keep file I/O at a minimum. I think we can safely assume that most systems should be able to keep 75MB objects in memory.